ocs_ci/ocs/platform_nodes.py

```diff
@@ -266,37 +266,54 @@ def restart_nodes(self, nodes, force=True, timeout=300, wait=True):
         Args:
             nodes (list): The OCS objects of the nodes
             force (bool): True for Hard reboot, False for Soft reboot
-            timeout (int): time in seconds to wait for node to reach 'not ready' state,
-                and 'ready' state.
+            timeout (int): time in seconds to wait for node to reach READY state
             wait (bool): True if need to wait till the restarted OCP node
                 reaches READY state. False otherwise
 
         """
+        num_events_pre_reboot, num_events_post_reboot = ([] for i in range(2))
         vms = self.get_vms(nodes)
+        node = nodes[0]
         assert vms, (
             f"Failed to get VM objects for nodes {[n.name for n in nodes]}"
         )
+
+        reboot_events_cmd = (
+            f"get events -A --field-selector involvedObject.name="
+            f"{node.name},reason=Rebooted -o yaml"
+        )
+
+        for node in nodes:
+            num_events_pre_reboot.append(
+                len(node.ocp.exec_oc_cmd(reboot_events_cmd)['items'])
+            )
+
         self.vsphere.restart_vms(vms, force=force)
 
+        logger.info("Waiting for 30 seconds...")
+        time.sleep(30)
+
         if wait:
             """
             When reboot is initiated on a VM from the VMware, the VM
             stays at "Running" state throughout the reboot operation.
 
-            Once the OCP node detects that the node is not reachable then the
-            node reaches status NotReady.
             When the reboot operation is completed and the VM is reachable the
-            OCP node reaches status Ready.
+            OCP node reaches status Ready and a Reboot event is logged.
             """
             nodes_names = [n.name for n in nodes]
-            wait_for_nodes_status(
-                node_names=nodes_names, status=constants.NODE_NOT_READY,
-                timeout=timeout
-            )
+
             wait_for_nodes_status(
                 node_names=nodes_names, status=constants.NODE_READY,
                 timeout=timeout
             )
+            for node in nodes:
+                num_events_post_reboot.append(
+                    len(node.ocp.exec_oc_cmd(reboot_events_cmd)['items'])
```

Comment:
Modify node name in reboot_events_cmd in each iteration

Reply:
Done